### PR TITLE
Fix #4684: Revert (most of) #4587

### LINKED ISSUE #4684: Promoted story cards fit images rather than constraining them
**Describe the bug**
Promoted story thumbnail cards do not consistently size thumbnail image sizes.

**To Reproduce**
1. Open a tip-of-tree version of the app with production assets.
2. Navigate to the home screen
3. Ensure that "What is Addition?" shows up in the top recommended stories.
4. Observe that the "What is Addition?" thumbnail is much larger than others.

**Expected behavior**
Promoted story card thumbnails should be consistently sized and not dependent on their physical sizes.

**Demonstration**
Screenshot:

image

**Environment**
- Device/emulator being used: Nexus 5X
- Android or SDK version (e.g. Android 5 or SDK 21): SDK 27
- App version (you can get this through system app settings or via the admin controls menu in-app): developer build using e4e4de63e0ade2a1f15e961be447dc3bf6d12eb0.

**Additional context**
Note this requires production assets--the local developer assets don't seem to trigger the issue.

### PR DESCRIPTION
**Pull request by BenHenning**
## Explanation
Fixes #4684

This PR reverts #4587. I think the change of match_parent to wraps_content caused the promoted story container to grow for larger images, and this wasn't caught until we tried the changes with production assets. I noticed other places in the PR that might have the same issue, so rather than trying to make sense of everything in order to properly fix it & add tests, I opted to revert the change since the regression is beta-blocking (even though it wasn't a clean reversion and required manual changes).

Note that the reversion includes keeping the changes to recently_played_fragment since otherwise changes in RecentlyPlayedFragmentTest fail, and the change seems unlikely to cause any problems with respect to potential thumbnail sizing issues.

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
There's not much to show here--see the issue for context on the problem. The post-reversion just reverts things back to their previous state (i.e. no regression).

### CODE REVIEW

File: app/src/main/res/layout/exploration_activity.xml
Comment (BenHenning) [inline app/src/main/res/layout/exploration_activity.xml]: Wrong

Comment (BenHenning) [reply app/src/main/res/layout/exploration_activity.xml]: Fixed in the latest commit.

**Comment by BenHenning** [comment]
@seanlip PTAL. Since this is mostly a clean reversion, I think it's fine to submit without @rt4914's review (plus I suspect he'll look at it with Vraj given the base PR). Note that CI passed before syncing to latest develop, so probably don't need to block on that for approval.

File: app/src/main/res/layout/app_language_activity.xml
Comment (seanlip) [inline app/src/main/res/layout/app_language_activity.xml]: We should drop this space, right?

**Review by seanlip — APPROVED** [review]
Just one note? Otherwise LGTM

**Comment by vrajdesai78** [comment]
@BenHenning I am good with the revert of changes related to thumbnail image. I have made that change previously because accessibility scanner is giving a suggestion to improve scaling with "Fixed witdh"

image

File: app/src/main/res/layout/app_language_activity.xml
Comment (BenHenning) [inline app/src/main/res/layout/app_language_activity.xml]:
```suggestion
      app:layout_constraintTop_toBottomOf="@id/reading_list_app_bar_layout">
```

Comment (BenHenning) [reply app/src/main/res/layout/app_language_activity.xml]: Yeah I can do that--it was originally in the file so it was copied over during reversion.

Comment (BenHenning) [reply app/src/main/res/layout/app_language_activity.xml]: (Marking as resolved since this is a trivial change and you've already approved @seanlip).

**Comment by BenHenning** [comment]
Thanks @seanlip! Enabling auto-merge for once CI finishes.

@vrajdesai78 we'll need to find a way to satisfy the constraints such that it doesn't break the app (wrap_content isn't the correct thing to do here).

**Comment by BenHenning** [comment]
Ah forgot that I need to force-merge since technically this requires @rt4914's codeowners approval.

**Pull request merged**
